### DISCUSSION TITLE
Upgrade old version quasar project to latest version

### DISCUSSION THREAD

**Discussion by narendraparihar** [quasarframework/quasar#13545] [CLI - PWA mode]
I am facing problems while upgrading my large-scale quasar project to the latest version. Is there any better way to upgrade the version of the project? The migration guide given on quasar's official website is very lengthy and it doesn't give surety whether the project will be completely upgraded or not.  

Suggest me any better way to upgrade a large-scale project in a lesser amount of time and the core functionality of the project remain intact

**Comment by MilosPaunovic** [comment]
[Upgrade guide](https://quasar.dev/start/upgrade-guide) on the official documentation page is pretty comprehensive and it covers all things Quasar; you should read it and follow it. What could be the issue in a large project are non-Quasar libraries which might not have support for Vue 3.

**Comment by robert-niestroj** [comment]
Have also a big project. Quasar isn't the biggest issue. It's 3rd party libs which sometimes are not available for Vue3 or the api changes so much that you can throw it out of the window.

**Comment by metalsadman** [comment]
Create a new project, copy paste your legacy code, read upgrade guide, use ide search and replace function for multiple edits of a keyword ie. `:value @input` to `:model-value @update:model-value`, test until error logs/lints are gone. Take your time, heed advices above, if it's a big project, then don't expect that it will run smoothly in a day. Took me a week or so(on and off) to migrate a fairly large quasar v1 codebase to v2, codebase was written originally since quasar v0.14,  been following upgrade guides since then so it's  quite `surety` :p at least in my experience.
